### PR TITLE
Add nightly rebase

### PR DESCRIPTION
**Pull request by fzdarsky**
Adds a periodic job to attempt a rebase of MicroShift onto the latest OpenShift nightly.

### CODE REVIEW

File: ci-operator/jobs/openshift/microshift/openshift-microshift-main-periodics.yaml [hunk 1]
```diff
@@ -1021,6 +1105,62 @@ periodics:
     ci-operator.openshift.io/variant: periodics
     ci.openshift.io/generator: prowgen
     pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-openshift-microshift-main-periodics-rebase-on-nightlies
```
Comment (dhellmann) [inline ci-operator/jobs/openshift/microshift/openshift-microshift-main-periodics.yaml:1108]: Eventually we'll have 1 of these jobs for each supported branch, right? Should the version number be included in the name?

Comment (fzdarsky) [reply ci-operator/jobs/openshift/microshift/openshift-microshift-main-periodics.yaml:1108]: The pattern seems to be to include the branch name into the job + job file name. So guess "main" is correct here and we'd add a "release-4.x" version once we create a corresponding branch.

**Comment by fzdarsky** [comment]
USHIFT-82

**Comment by fzdarsky** [comment]
USHIFT-179

File: ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  from: src
+  cli: nightly
+  grace_period: 30m
+  commands: openshift-microshift-rebase-commands.sh
+  resources:
+    requests:
+      cpu: 100m
+      memory: 100Mi
+  dependencies:
+  - name: "release:nightly-amd64"
```
Comment (zshi-redhat) [inline ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-ref.yaml:12]: Should this be `release:nightly`? I didn't see `release:nightly-amd64` elsewhere in the repo, but there is `release:nightly-arm64`.

File: ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-commands.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+oc get configmaps
+
+sleep 1800
+
+# fetch pull-secret for the central CI registry
+mkdir -p ~/.docker
+oc registry login --registry-config="${HOME}/.docker/config.json"
+
+TARGET_RELEASE_IMAGE="quay.io/openshift-release-dev/ocp-release:4.10.18-x86_64"
```
Comment (zshi-redhat) [inline ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-commands.sh:26]: Why not using `TARGET_RELEASE_AMD64`?

File: ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-ref.yaml (hunk 2 above)
Comment (zshi-redhat) [reply ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-ref.yaml:12]: oh, I see you defined the nightly-amd64 in *main__periodics.yaml

File: ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-commands.sh (hunk 3 above)
Comment (fzdarsky) [reply ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-commands.sh]: That was an experiment. Please see the updated PR.

**Comment by ggiguash** [comment]
/lgtm
Note that the implementation has not been completed. We require to merge this PR before the next steps.

**Pull request merged**